### PR TITLE
[BBC-12726] Node 15 + NPM 7 + update peer deps

### PR DESCRIPTION
**Pull request by mcellier**
## Description

Upgrade to node 15 / npm 7 to match the target version before intenalizing the lib.

## Things to consider

Also updated peer deps numbers to match kairos.

## How it was tested

Canary release by @guillaumewuip

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -8,8 +8,8 @@
     "url": "git@github.com:blablacar/ui-library.git"
   },
   "engines": {
-    "node": "^12.16.1",
-    "npm": "^6.13.4"
+    "node": "^15.11.0",
+    "npm": "^7.6.0"
```
Comment (guillaumewuip) [inline package.json:12]:
```suggestion
    "node": "^12.16.1 | ^15.11.0",
    "npm": "^6.13.4 | ^7.6.0"
```

Maybe this to be backward compatible?

Comment (mcellier) [reply package.json:12]: Trying this syntax but it doesn't seems to work. 
I got the following working:

```
"node": ">=12.16.1 <=v15.11.0",
"npm": ">=6.13.4 <=7.6.0"
```

WDYT?

Comment (mcellier) [reply package.json]: 👌

Comment (mcellier) [reply package.json]: You're right. Moreover, the `.nvmrc` ensure the version so we're good IMHO.

**Review by guillaumewuip — APPROVED** [review]

**Pull request merged**